Documentation/generics.md

```diff
@@ -0,0 +1,64 @@
+*.NET Innovation Examples*
+**GENERICS**
+===================
+
+We use generics implicitly all the time in  C#. When you use LINQ in C#, do you notice you are working with <code>IEnumerable\<T\></code>? Ever wonder what <code>T</code> is and what it means?
+
+First introduced to the .NET Framework 2.0, generics involved changes to both the C# language and the Common Language Runtime (CLR). **Generics** are essentially a "code template" that allows developers define [type-safe](https://msdn.microsoft.com/en-us/library/hbzz1a9a%28v=vs.110%29.aspx) data structures without committing to an actual data type. For example, <code>List\<T\></code> is a [Generic Collection](https://msdn.microsoft.com/en-us/library/System.Collections.Generic(v=vs.110).aspx) that can be declared and used with any type: <code>List\<int\></code>, <code>List\<string\></code>, <code>List\<Person\></code>, etc.  
+
+So, what's the point? Why are generics useful? The general idea in C# is that developers should have a **strongly-typed experience** with data structures. 
+> A [strongly-typed ](https://msdn.microsoft.com/en-us/library/ms173104.aspx) experience means that the language predefines types of data and every variable or constant has to be described as one of those predefined types. This is also referred to as **"type-safe."** You can read more about the importance of type-safety [here](http://www.gibraltarsoftware.com/vistadb/what-you-get/technical/managed-code).
```

Comment:
You might also be able link here: https://github.com/dotnet/coreclr/blob/master/Documentation/intro-to-clr.md#memory-and-type-safety